File: examples/dotnet/FormSubmissionIntegrityVerifier.cs
using System.Security.Cryptography;
using System.Text;

namespace dotnet
{
  public class FormSubmissionIntegrityVerifier
  {
    public static bool VerifyIntegrity(string answers, string checksum)
    {
      try
      {
        byte[] expectedHashSequence = Convert.FromHexString(checksum);

        byte[] receivedData = Encoding.UTF8.GetBytes(answers);
        byte[] hashSequenceFromReceivedData = MD5.HashData(receivedData);

        return hashSequenceFromReceivedData.SequenceEqual(expectedHashSequence);
      }
      catch (Exception exception)
      {
        throw new Exception("Failed to verify integrity", exception);
      }
    }
  }
}

File: examples/python/form_submission_integrity_verifier.py
import hashlib


class FormSubmissionVerifier:

    def verify_integrity(answers: str, checksum: str) -> bool:
        try:
            expected_hash_sequence = bytes.fromhex(checksum)

            received_data = answers.encode("utf-8")
            hash_sequence_from_received_data = hashlib.md5(received_data).digest()

            return hash_sequence_from_received_data == expected_hash_sequence
        except Exception as exception:
            raise Exception("Failed to verify integrity") from exception
